### PR TITLE
Extend subgroup #160

### PR DESCRIPTION
**Pull request by elong0527**
The PR show the possibility to avoid `prepare_ae_specific_subgroup`, `format_ae_specific_subgroup` functions by creating internal functions to handle subgroup analysis. 

example: 
```
devtools::load_all()

meta <- meta_ae_example()
tbl <- prepare_ae_specific(meta,
  population = "apat",
  observation = "wk12",
  parameter = "rel"
) |>
  extend_ae_specific_events() |> 
  extend_ae_specific_duration(duration_var = "ADURN") |>
  extend_ae_specific_subgroup(subgroup = "SEX") |>
  format_ae_specific()

tbl_subgroup <- outdata_eval_prepare_call(outdata_population_subgroup(tbl, tbl$subgroup))
```

The `tbl_subgroup` provide subgroup analysis. For example: 

```
> tbl$n_pop
  n_1 n_2 n_3 n_4
1  86  84  84 254
> tbl_subgroup[[1]]$n_pop
  n_1 n_2 n_3 n_4
1  53  50  40 143
> tbl_subgroup[[2]]$n_pop
  n_1 n_2 n_3 n_4
1  33  34  44 111
```

The general idea is: 

1. add `prepare_call` and `extend_call` into `outdata` object to remember the exact call used in AE specific analysis. 
2. update `meta$population` to add subgroup population using `outdata_population_subgroup` 
3. execute the same AE specific analysis among each subgroup using `outdata_eval_prepare_call`. 

If we like the idea, we need to modify the `tlf_ae_specific` and `tlf_ae_specific_subgroup`:

1. combine `outdata$tbl` from the overall and each subgroup `outdata`. 
2. generate RTF files in desired format.

### CODE REVIEW

**Comment by elong0527** [comment]
Please keep the PR open. Will need to discuss with @wangben718 to avoid duplicated code.

File: R/extend_ae_specific.R [hunk 1]
```diff
@@ -333,6 +334,18 @@ extend_ae_specific_events <- function(outdata) {
...
+  outdata$extend_call <- c(outdata$extend_call, match.call())
+
+  outdata
+}
+
+#' Add Subgroup analysis in AE specific analysis
+#'
+#' @param subgroup a character string for subgroup variable name
+#'
+#' @export
```
Comment (nanxstats) [inline R/extend_ae_specific.R:346]: - Should "subgroup" in the title be in lower case?
- `@params` description should use sentence case with a full stop.
- Exported functions should have `@return` and `@examples` sections.

**Comment by elong0527** [comment]
User will not aware of this. Those 'outdata_xxx' function can be embedded into the 'tlf_ae_specific' function. The only new function user need to learn is 'extend_ae_specific_subgroup'.

**Comment by wangben718** [comment]
Let me merge it and I will take over the future development

**Review by wangben718 — APPROVED** [review]

**Pull request merged**